packages/clerk-js/src/utils/url.ts

```diff
@@ -350,3 +350,22 @@ export const isAllowedRedirectOrigin = (_url: string, allowedRedirectOrigins: Ar
   }
   return isAllowed;
 };
+
+export function createAllowedRedirectOrigins(
+  allowedRedirectOrigins: Array<string | RegExp> | undefined,
+  frontendApi: string,
+): (string | RegExp)[] | undefined {
+  if (!allowedRedirectOrigins || allowedRedirectOrigins.length === 0) {
+    const origins = [];
+    if (typeof window !== 'undefined' && !!window.location) {
+      origins.push(window.location.origin);
+    }
+
+    origins.push(`https://${getETLDPlusOneFromFrontendApi(frontendApi)}`);
+    origins.push(`https://*.${getETLDPlusOneFromFrontendApi(frontendApi)}`);
+
+    return origins;
+  }
+
+  return allowedRedirectOrigins;
```

Comment:
Nitpicky, feel free to ignore. I usually like to flip the conditional and write this the other way around: 

```
if (Array.isArray(allowedRedirectOrigins) && !!allowedRedirectOrigins.length) {
  return allowedRedirectOrigins;
} 
// more logic
```

I find it slightly more readable as you take care of the simple things right away and then handle the "difficult" part

Reply:
Agree, Done!